SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -77,6 +104,19 @@ SpatialEventTracer::~SpatialEventTracer()
 	}
 }
 
+FString SpatialEventTracer::SpanIdToString(const Trace_SpanId& SpanId)
+{
+	FString HexStr;
+	for (int i = 0; i < 16; i++)
+	{
+		char b[32];
+		unsigned int x = (unsigned char)SpanId.data[i];
+		sprintf(b, "%0x", x);
```

Comment:
can this instead just be 
```
char b;
sprintf(&b, "%x", spanIdData[i]);
```

Then you trim the 0s from the string at the end?

Reply:
I'm slightly worried that using 0 in the sprintf is going to give garbage in the case where the result is 0. As I'm not sure if that just leaves b uninitialized I think.
Similarly b is a 32 byte array but this only uses the first element.

Reply:
Why would b be uninitialized if SpanId.data[i] == 0? Surely we would want to represent 0x00 and the print function should allow that. Perhaps I miss understood. 

I do have valid 0 SpanId string:
LogSpatialEventTracer: SpanId: eff950268f144839632bd26cbe456722 Causes: 00000000000000000000000000000000, 00000000000000000000000000000000

Reply:
Think %02x is probably desired here since the size won't exceed 2 chars (for a 32 char 0-padded hex string)

Reply:
So b is uninitialized and I can't tell from https://en.cppreference.com/w/cpp/io/c/fprintf if sprintf option 3 has to write a null terminator to the buffer in the event that it has no characters to write. In fact I'm not 100% certain if it ensures the string is null terminated if it does have characters. I might just be misreading it though.
(I'm mostly worried because option 4 does explicitly say that it will result in something null terminated regardless. So maybe it's safer just to use that).

@improbable-danny yeah I think actually you do always want .2 precision as each byte = two characters. I don't know why I thought it was one earlier.
however it does still look like you should be able to get rid of x as a temporary.

Reply:
Was having some warning in buildkite using sprintf. Using FString::Printf(TEXT("%02x"), SpanId.data[i]) which seems legit.

Reply:
snprintf might be a better option - doesn't require dynamic allocation! Maybe
`if(snprintf(b, sizeof(b), "%02x", (unsigned integer)SpanId.data[i]) > 0`

Worth noting %x will read an unsigned integer so needs to be 0-padded (FString::Printf I think does some fancy template stuff to be type safe though). Think snprintf will always null-terminate if size > 1, at least the glibc implementation appears to (only ever writes n-1 characters)

Reply:
I think I will merge this version for now as this function is only used for debugging/tests. My work on M3 will be touching this code. I will see if we can avoid the allocation there.